tdrs-backend/tdpservice/parsers/schema_defs/tanf/t2.py

```diff
@@ -11,145 +11,370 @@
+            validators.if_then_validator(
+                  condition_field='FAMILY_AFFILIATION', condition_function=validators.matches(1),
+                  result_field='SSN', result_function=validators.notOneOf(("000000000", "111111111", "222222222",
+                                                                           "333333333", "444444444", "555555555",
+                                                                           "666666666", "777777777", "888888888",
+                                                                           "999999999")),
+            ),
+            validators.if_then_validator(
+                  condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                  result_field='RACE_HISPANIC', result_function=validators.oneOf((1, 2)),
+            ),
+            validators.if_then_validator(
+                      condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                      result_field='RACE_AMER_INDIAN', result_function=validators.oneOf((1, 2)),
+                ),
+            validators.if_then_validator(
+                      condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                      result_field='RACE_ASIAN', result_function=validators.oneOf((1, 2)),
+                ),
+            validators.if_then_validator(
+                      condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                      result_field='RACE_BLACK', result_function=validators.oneOf((1, 2)),
+                ),
+            validators.if_then_validator(
+                      condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                      result_field='RACE_HAWAIIAN', result_function=validators.oneOf((1, 2)),
+                ),
+            validators.if_then_validator(
+                      condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                      result_field='RACE_WHITE', result_function=validators.oneOf((1, 2)),
+                ),
+            validators.if_then_validator(
+                      condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                      result_field='MARITAL_STATUS', result_function=validators.oneOf((1, 2, 3, 4, 5)),
+                ),
+            validators.if_then_validator(
+                      condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                      result_field='PARENT_WITH_MINOR_CHILD', result_function=validators.isInLimits(1, 3),
+                ),
+            validators.if_then_validator(
+                      condition_field='FAMILY_AFFILIATION', condition_function=validators.isInLimits(1, 3),
+                      result_field='EDUCATION_LEVEL', result_function=validators.notMatches(0),
```

Comment:
if `FAMILY_AFFILIATION` == 1 and `EDUCATION_LEVEL` == 17 (which would be an out of range value, would this validator catch the error?  is `notMatches` the correct validator for this check?

Reply:
The error in excel file is: "If item 30 == 1, 2, or 3, then item 41 == 01-16,98 or 99", with the Field validator covering the whole range, only zero is left to cover. Nevertheless, we changed the validator